test/README.md

```diff
@@ -0,0 +1,66 @@
+# Tests Overview
+
+## Full Invariant List
+
+1. `CoinbaseSmartWallet`
+   1. Only transaction path impacted is when `PermissionManager` is used as owner
+   2. Existing transaction paths work exactly the same
+      1. Can transact using other owners
+      2. Can remove `PermissionManager` as an owner
+      3. Can add and remove other owners
+      4. Can upgrade the contract
+2. `PermissionManager`
+   1. Can only validate signatures for ERC-4337 User Operations
+      1. Only supports the same UserOp type as `CoinbaseSmartWallet` (v0.6)
+   2. Cannot update `CoinbaseSmartWallet` owners
+   3. Cannot upgrade `CoinbaseSmartWallet` implementation
+   4. Cannot make self-calls on `CoinbaseSmartWallet` as a precaution
+   5. Only returns User Operations as valid when:
+      1. `PermissionManager` is not paused
+      2. UserOp is signed by Permission signer ("session key")
+      3. UserOp is signed by cosigner or pending cosigner (managed by Coinbase)
+      4. UserOp paymaster is enabled by `PermissionManager`
+      5. UserOp and Permission are validated together by Permission Contract
+      6. Permission Contract is enabled by `PermissionManager`
+      7. Permission account matches UserOp sender
+      8. Permission chain matches current chain
+      9. Permission verifyingContract is `PermissionManager`
+      10. Permission has not expired
+      11. Permission is approved via storage or signature by a different owner on smart wallet
```

Comment:
```suggestion
      11. Permission is approved via storage or signature by any owner on smart wallet
```

I think it doesn't have to be by a different owner, it could be any owner?

Reply:
It could be any owner, but I think it's important for the manager to not self-authorize. My intuition is that it should come from another owner to initialize the permission for checks & balances (I think). The design of `isValidSignature` only parsing userOps is one way to effectively guarantee this.